packages/frontend/src/redux/slices/account/createAccountThunks.js

```diff
@@ -0,0 +1,176 @@
+    }, { dispatch }) => {
+        if (recoveryMethod === 'ledger') {
+            await wallet.addLedgerAccountId(accountId);
+            await wallet.postSignedJson('/account/ledgerKeyAdded', { accountId, publicKey: publicKey.toString() });
+        } else {
+            const newKeyPair = KeyPair.fromRandom('ed25519');
+            const newPublicKey = newKeyPair.publicKey;
+            if (recoveryMethod !== 'phrase') {
+                await wallet.addNewAccessKeyToAccount(accountId, newPublicKey);
+                await wallet.saveAccount(accountId, newKeyPair);
+            } else {
+                const contractName = null;
+                const fullAccess = true;
+                await wallet.postSignedJson('/account/seedPhraseAdded', { accountId, publicKey: publicKey.toString() });
+                try {
+                    await wallet.addAccessKey(accountId, contractName, newPublicKey, fullAccess);
+                    await wallet.saveAccount(accountId, newKeyPair);
+                } catch (error) {
+                    if (previousAccountId) {
+                        await wallet.saveAndMakeAccountActive(previousAccountId);
+                    }
+                    throw new WalletError(error, 'addAccessKey.error');
+                }
+            }
+        }
+
+        setReleaseNotesClosed(RELEASE_NOTES_MODAL_VERSION);
+        await dispatch(finishAccountSetup());
+    }
+);
+
+export const createIdentityFundedAccount = createAsyncThunk(
+    `${SLICE_NAME}/createIdentityFundedAccount`,
+    async ({
+        accountId,
+        kind,
+        publicKey,
+        identityKey,
+        verificationCode,
+        recoveryMethod,
+        recaptchaToken,
+        recaptchaAction
+    }, { dispatch }) => {
+        await wallet.checkNewAccount(accountId);
+        await sendJson('POST', IDENTITY_FUNDED_ACCOUNT_CREATE_URL, {
+            kind,
+            newAccountId: accountId,
+            newAccountPublicKey: publicKey.toString(),
+            identityKey,
+            verificationCode,
+            recaptchaToken,
+            recaptchaAction,
+            recaptchaSiteKey: RECAPTCHA_ENTERPRISE_SITE_KEY
+        });
+        await wallet.saveAndMakeAccountActive(accountId);
+        await dispatch(addLocalKeyAndFinishSetup({ accountId, recoveryMethod, publicKey }));
+    }
+);
+
+export const createNewAccount = createAsyncThunk(
+    `${SLICE_NAME}/createNewAccount`,
+    async ({
+        accountId,
+        fundingOptions,
+        recoveryMethod,
+        publicKey,
+        previousAccountId,
+        recaptchaToken
+    }, { dispatch }) => {
+        await wallet.checkNewAccount(accountId);
+
+        const { fundingContract, fundingKey, fundingAccountId } = fundingOptions || {};
+        if (fundingContract && fundingKey) {
+            await wallet.createNewAccountLinkdrop(accountId, fundingContract, fundingKey, publicKey);
+            await wallet.keyStore.removeKey(NETWORK_ID, fundingContract);
```

Comment:
I see this is a pattern we've been using elsewhere for a while but it would be nice at some point to wrap this behind a method in `wallet` (e.g. `await wallet.removeNetworkKey(...)` or something) to avoid exposing too much and protect against future implementation changes in `wallet`. Just thinking out loud, not something that needs to be addressed here 🙂

Reply:
That is a good idea.